ldap_validator/ldap_validator.py

```diff
@@ -0,0 +1,474 @@
+#     usage()     #
+###################
+def usage(exit_code: int):
+    """
+    Prints the usage information for the pre-install-report command and exits the program with the given exit_code.
+
+    :param exit_code: The exit code to return when exiting the program.
+    """
+    print("\n")
+    print("Usage: python3 ldap_validator.py <-s|--sitedefault> [<options>]\n")
+    print("Example: python3 ldap_validator.py -s /path/to/sitedefault.yaml\n")
+
+    print("Options:\n")
+    print("    -s  --sitedefault           (Required)Full path the sitedefault.yaml file being validated.")
+    print("    -o, --output-dir=\"<dir>\"    (Optional)Write the report and log files to the provided directory")
+    print("    -d, --debug                 (Optional)Enables logging at DEBUG level. Default is INFO level")
+    print("    -h  --help                  (Optional)Show this usage message")
+    sys.exit(0)
+    print()
+    sys.exit(exit_code)
+
+
+# --------------------------------------------------------------------------------------------------
+##################
+#     main()     #
+##################
+def main(argv):
+
+    """
+    Main execution point for the ldap_validator script
+
+    Argument:
+        arguments(string) - The parameters passed to the script at execution.
+    """
+    global sas_logger
+
+    sitedefault_loc = ""
+    output_dir = ""
+    logging_level: int = logging.INFO
+
+    # Parse command line options/arguments
+    try:
+        opts, args = getopt.getopt(argv, "s:o:dh", ["sitedefault=", "output-dir=", "debug", "help"])
+    except getopt.GetoptError as opt_error:
+        print(opt_error)
+        usage(ldap_messages.BAD_OPT_RC_)
+
+    for opt, arg in opts:
+
+        if opt in ('-s', '--sitedefault'):
+            sitedefault_loc = arg
+        elif opt in ('-o', '--output-dir'):
+            output_dir = arg
+        elif opt in ('-d', '--debug'):
+            print("Setting log level to DEBUG  ")
+            logging_level = logging.DEBUG
+        elif opt in ('-h', '--help'):
+            usage(ldap_messages.SUCCESS_RC_)
+        else:
+            print()
+            print(ldap_messages.OPTION_ERROR.format(str(opt)), 'error')
+            usage(ldap_messages.BAD_OPT_RC_)
+
+        # make sure path is valid #
+    if output_dir != "":
+        if not output_dir.endswith(os.sep):
+            output_dir = output_dir + os.sep
+        else:
+            print(ldap_messages.OUPUT_PATH_ERROR)
+            usage(ldap_messages.BAD_OPT_RC_)
+
+        # Set the full log path
+    validator_log_path = output_dir + _VALIDATOR_LOG_NAME_TMPL_.format(file_timestamp)
+
+    try:
+        logging.FileHandler(validator_log_path)
+    except OSError as e:
+        print(ldap_messages.OUPUT_PATH_ERROR, format(e))
+        usage(ldap_messages.BAD_OPT_RC_)
+
+    sas_logger = ViyaARKLogger(validator_log_path, logging_level=logging_level, logger_name="ldap_validator_logger")
+    ldap_logger = sas_logger.get_logger()
```

Comment:
Now I see association of ldap_logger var with sas_logger.     if ldap_logger also declared as global would allow you to remove repeated input of ldap_logger to other methods.

Reply:
Yes and no.  Setting the logger variables to global and not passing it to the functions is a cleaner way of doing this, but it doesn't seem to work with Pytest. We can discuss during the review. I would prefer to do it the way you mentioned, and that is how I had it initially, but Pytest doesn't seem to pick up anything from main. If there is a way to get Pytest to pick up the variable settings from main, or a better way to do it that would be great. We can discuss and brainstorm in the review meeting.